### PR TITLE
App

### PR DESCRIPTION
**Pull request by wesleyjia34**
The following is currently completed:

- Initial draft of all reducers
- User-related action creators
- Action types
- Component structures for the login page, create user page, and main application, with some placeholder text/HTML elements representing what will actually be placed in each component

### CODE REVIEW

File: client/actions/userActionCreators.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
+import actions from '../constants/actions.js';
+
+export const checkSessionActionCreator = () => (dispatch) => {
+/*
+  thunked action creator that sends request to check
```
Comment (Peter-Ba) [inline client/actions/userActionCreators.js:5]: what is thunked

File: client/actions/userActionCreators.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+export const userLoginActionCreator = (e) => (dispatch) => {
+/*
+  thunked action creator expecting argument (e) 
+  representing a React synthetic event for login form submission
+*/
+
+  e.preventDefault();
+
+  fetch('https://localhost:3000/api/users/create', {
```
Comment (wesleyjia34) [inline client/actions/userActionCreators.js]: users/login

File: client/actions/userActionCreators.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({
+      userName: e.target[0].value,
```
Comment (wesleyjia34) [inline client/actions/userActionCreators.js]: username lower case

File: client/actions/userActionCreators.js (hunk 1 above)
Comment (wesleyjia34) [reply client/actions/userActionCreators.js:5]: thunking (in general) refers to delaying the execution of a function. In this case, it's specifically for use with the 'redux-thunk' middleware (which is an installed package).

redux-thunk allows us to use async operations in our action creators. This allows us make a HTTP request before we dispatch an action, for example. (without thunking, there is no way to do an async operation before dispatching an action)

**Review by odylic — APPROVED** [review]
Looks good

**Pull request merged**